### DISCUSSION TITLE
Leverage SharePoint REST API

### DISCUSSION THREAD

**Discussion by advocate-matt-joers** [microsoft-search/pnp-modern-search#2620] [Ideas]
**Is your feature request related to a problem? Please describe.**
Has there been any appetite to incorporate the SharePoint List / Library REST API natively within the modern search functionality?

**Describe the solution you'd like**
The idea I've been toying with is to populate the list results with data from the SharePoint REST API, in addition to allowing users to use the REST API to target specific list fields as filter values as refiners.

**Describe alternatives you've considered**
This would be an alternative to using the MS Search or classic SharePoint Search which would avoid the need to promote refinablestrings among other customizations needed from the SharePoint Search configuration settings.

**Additional context**

**Comment by edarroudi** [comment]
You can create a [custom data source](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_data_sources/).
Adding a standard REST API DataSource (in the main solution) was declined. There is a solution that you can purchase. [See](https://github.com/microsoft-search/pnp-modern-search/discussions/1471#discussioncomment-1581624) .

I'm working on a generic solution that can access any datasource including auth scenarious with oauth, so Azure and SP connections would be possible... But that will take some time (early alpha) and I'm not sure in which repository it will get published.

**Comment by FranckyC** [comment]
Hi @advocate-matt-joers, this has been discussed in the past and as the PnP Modern Search is focused on search, this use case deviates a little it from the original purpose. However since we released in 4.8 the custom data sources as extensions, feel free to submit this data source to the extensibility sample repository: https://github.com/microsoft-search/pnp-modern-search-extensibility-samples
